File: src/main/java/maxdistructo/droidbot2/commands/mafia/MafiaConfig.java
package maxdistructo.droidbot2.commands.mafia;

import maxdistructo.droidbot2.core.Roles;
import maxdistructo.droidbot2.core.Utils;
import maxdistructo.droidbot2.core.message.Message;
import org.apache.commons.lang3.ArrayUtils;
import org.json.JSONArray;
import org.json.JSONObject;
import sx.blah.discord.handle.obj.IMessage;
import sx.blah.discord.handle.obj.IUser;

import java.io.File;
import java.io.FileWriter;
import java.io.IOException;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.List;

public class MafiaConfig {
    public static long getAdminChannel(IMessage message) {
        JSONObject root = Utils.readJSONFromFile("/config/mafia/" + message.getGuild().getLongID() + "_dat.txt");
        return root.getLong("adminChannel");
    }

    public static long getDayChat(IMessage message) {
        JSONObject root = Utils.readJSONFromFile("/config/mafia/" + message.getGuild().getLongID() + "_dat.txt");
        return root.getLong("dayChat");
    }

    public static long[] getPlayers(IMessage message) {
        JSONObject root1 = Utils.readJSONFromFile("/config/mafia/" + message.getGuild().getLongID() + "_players.txt");
        JSONArray jsonArray = root1.getJSONArray("players");
        long[] strArray = new long[jsonArray.length()];
        for (int i = 0; i < jsonArray.length(); i++) {
            strArray[i] = jsonArray.getLong(i);
        }
        return strArray;
    }

    public static long[] getPlayers(IMessage message, String role) {
        List<IUser> usersList = message.getGuild().getUsersByRole(Roles.getRole(message, role));
        usersList.remove(message.getClient().getOurUser());
        long[] players = new long[usersList.size()];
        int i = 0;
        for (IUser user : usersList) {
            players[i] = user.getLongID();
            i++;
        }
        return players;
    }

    public static long getMafiaChannel(IMessage message) {
        JSONObject root1 = Utils.readJSONFromFile("/config/mafia/" + message.getGuild().getLongID() + "_dat.txt");
        return root1.getLong("mafia_chat");
    }

    public static boolean getDayStatus(IMessage message) {
        JSONObject root1 = Utils.readJSONFromFile("/config/mafia/" + message.getGuild().getLongID() + "_dat.txt");
        return root1.getBoolean("day");
    }

    public static Object[] getPlayerDetails(IMessage message) {
        JSONObject root1 = Utils.readJSONFromFile("/config/mafia/" + message.getGuild().getLongID() + "_playerdat.txt");
        JSONObject root = root1.getJSONObject("" + message.getAuthor().getLongID());
        return new Object[]{root.getString("alignment"), root.getString("class"), root.getString("role"), root.getBoolean("dead"), root.getInt("attack"), root.getInt("defense"), root.getBoolean("blocked"), root.getBoolean("framed"), root.getBoolean("jailed")};
    }

    public static Object[] getPlayerDetails(IMessage message, long playerID) {
        JSONObject root1 = Utils.readJSONFromFile("/config/mafia/" + message.getGuild().getLongID() + "_playerdat.txt");
        JSONObject root = root1.getJSONObject("" + playerID);
        return new Object[]{root.getString("alignment"), root.getString("class"), root.getString("role"), root.getBoolean("dead"), root.getInt("attack"), root.getInt("defense"), root.getBoolean("blocked"), root.getBoolean("framed"), root.getBoolean("jailed")};
    }

    public static long getMafiaChat(IMessage message) {
        return getMafiaChannel(message);
    }

    public static long getMediumChat(IMessage message) {
        JSONObject root1 = Utils.readJSONFromFile("/config/mafia/" + message.getGuild().getLongID() + "_dat.txt");
        return root1.getLong("medium_chat");
    }

    public static long getSpyChat(IMessage message) {
        JSONObject root1 = Utils.readJSONFromFile("/config/mafia/" + message.getGuild().getLongID() + "_dat.txt");
        return root1.getLong("spy_chat");
    }

    public static long getDeadChat(IMessage message) {
        JSONObject root1 = Utils.readJSONFromFile("/config/mafia/" + message.getGuild().getLongID() + "_dat.txt");
        return root1.getLong("dead_chat");
    }

    public static String[] shuffleJSONArray(JSONArray jsonArray) {
        String[] list = new String[jsonArray.length()];
        for (int i = 0; i < jsonArray.length(); i++) {
            list[i] = jsonArray.getString(i);
        }
        ArrayUtils.shuffle(list);
        return list;
    }

    public static void writeGame(IMessage message, JSONObject object) {
        Path currentRelativePath = Paths.get("");
        String s = currentRelativePath.toAbsolutePath().toString();

        File file = new File(s + "/config/mafia/" + message.getGuild().getLongID() + "_playerdat.txt");
        if (!file.exists()) {
            try {
                file.createNewFile();
            } catch (IOException e) {
                Message.throwError(e, message);
            }
        }

        try (FileWriter fileWriter = new FileWriter(s + "/config/mafia/" + message.getGuild().getLongID() + "_playerdat.txt")) {
            fileWriter.write(object.toString());
            System.out.println("Successfully Copied JSON Object to File...");
            System.out.println("JSON Object: " + object);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }

    public static long getJailorChat(IMessage message) {
        JSONObject root1 = Utils.readJSONFromFile("/config/mafia/" + message.getGuild().getLongID() + "_dat.txt");
        return root1.getLong("jailor_chat");
    }

    public static long getJailedChat(IMessage message) {
        JSONObject root1 = Utils.readJSONFromFile("/config/mafia/" + message.getGuild().getLongID() + "_dat.txt");
        return root1.getLong("jailed_chat");
    }

    public static long getJailed(IMessage message) {
        JSONObject root = Utils.readJSONFromFile("/config/mafia/" + message.getGuild().getLongID() + "_playerdat.txt");
        return root.getLong("jailed");
    }
    public static void writeGameDat(IMessage message, JSONObject object){
        Path currentRelativePath = Paths.get("");
        String s = currentRelativePath.toAbsolutePath().toString();

        File file = new File(s + "/config/mafia/" + message.getGuild().getLongID() + "_dat.txt");
        if (!file.exists()) {
            try {
                file.createNewFile();
            } catch (IOException e) {
                Message.throwError(e, message);
            }
        }

        try (FileWriter fileWriter = new FileWriter(s + "/config/mafia/" + message.getGuild().getLongID() + "_dat.txt")) {
            fileWriter.write(object.toString());
            System.out.println("Successfully Copied JSON Object to File...");
            System.out.println("JSON Object: " + object);
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: src/main/java/maxdistructo/droidbot2/commands/mafia/MafiaConfig.kt
package maxdistructo.droidbot2.commands.mafia

import maxdistructo.droidbot2.core.Roles
import maxdistructo.droidbot2.core.Utils
import maxdistructo.droidbot2.core.message.Message
import org.apache.commons.lang3.ArrayUtils
import org.json.JSONArray
import org.json.JSONObject
import sx.blah.discord.handle.obj.IMessage
import java.io.File
import java.io.FileWriter
import java.io.IOException
import java.nio.file.Paths

object MafiaConfig {
    fun getPlayers(message: IMessage): LongArray {
        val root1 = Utils.readJSONFromFile("/config/mafia/" + message.guild.longID + "_players.txt")
        val jsonArray = root1.getJSONArray("players")
        val strArray = LongArray(jsonArray.length())
        for (i in 0 until jsonArray.length()) {
            strArray[i] = jsonArray.getLong(i)
        }
        return strArray
    }

    fun getPlayers(message: IMessage, role: String): LongArray {
        val usersList = message.guild.getUsersByRole(Roles.getRole(message, role))
        usersList.remove(message.client.ourUser)
        val players = LongArray(usersList.size)
        var i = 0
        for (user in usersList) {
            players[i] = user.longID
            i++
        }
        return players
    }

    fun getPlayerDetails(message: IMessage): Array<Any> {
        if(!Perms.checkMod(message)) {
            val root1 = Utils.readJSONFromFile("/config/mafia/" + message.guild.longID + "_playerdat.txt")
            val root = root1.getJSONObject("" + message.author.longID)
            val list = arrayListOf<Any>(root.getString("alignment"), root.getString("class"), root.getString("role"), root.getBoolean("dead"), root.getInt("attack"), root.getInt("defense"))
            return list.toArray()
        }
        else{
            val list = arrayListOf<Any>("admin", "admin", "admin", false, 3, 3)
            return list.toArray()
        }
    }

    fun getPlayerDetails(message: IMessage, playerID: Long): Array<Any> {
        if(!Perms.checkMod(message,playerID)) {
            val root1 = Utils.readJSONFromFile("/config/mafia/" + message.guild.longID + "_playerdat.txt")
            val root = root1.getJSONObject("" + playerID)
            val list = arrayListOf<Any>(root.getString("alignment"), root.getString("class"), root.getString("role"), root.getBoolean("dead"), root.getInt("attack"), root.getInt("defense"))
            return list.toArray()
        }
        else{
            val list = arrayListOf<Any>("admin", "admin", "admin", false, 3, 3)
            return list.toArray()
        }
    }

    fun shuffleJSONArray(jsonArray: JSONArray): Array<String?> {
        val list = arrayOfNulls<String>(jsonArray.length())
        for (i in 0 until jsonArray.length()) {
            list[i] = jsonArray.getString(i)
        }
        ArrayUtils.shuffle(list)
        return list
    }
    fun writeGame(message: IMessage, `object`: JSONObject) {
        val currentRelativePath = Paths.get("")
        val s = currentRelativePath.toAbsolutePath().toString()
        val file = File(s + "/config/mafia/" + message.guild.longID + "_playerdat.txt")
        if (!file.exists()) {
            try {
                file.createNewFile()
            } catch (e: IOException) {
                Message.throwError(e, message)
            }
        }
        try {
            FileWriter(s + "/config/mafia/" + message.guild.longID + "_playerdat.txt").use { fileWriter ->
                fileWriter.write(`object`.toString())
            }
        } catch (e: IOException) {
            e.printStackTrace()
        }

    }

    fun getJailed(message: IMessage): Long {
        val root = Utils.readJSONFromFile("/config/mafia/" + message.guild.longID + "_dat.txt")
        return root.getLong("jailed")
    }

    fun writeGameDat(message: IMessage, `object`: JSONObject) {
        val currentRelativePath = Paths.get("")
        val s = currentRelativePath.toAbsolutePath().toString()

        val file = File(s + "/config/mafia/" + message.guild.longID + "_dat.txt")
        if (!file.exists()) {
            try {
                file.createNewFile()
            } catch (e: IOException) {
                Message.throwError(e, message)
            }

        }

        try {
            FileWriter(s + "/config/mafia/" + message.guild.longID + "_dat.txt").use { fileWriter ->
                fileWriter.write(`object`.toString())
            }
        } catch (e: IOException) {
            e.printStackTrace()
        }

    }
}